keylime/cert_utils.py

```
@@ -67,6 +70,32 @@ def x509_der_cert(der_cert_data: bytes) -> Certificate:
 
+def unwrap_x509_octet_string(der_cert_data: bytes) -> bytes:
+    """If der_cert_data is an ASN.1 OCTET STRING wrapping a DER certificate,
+    unwrap it and return the inner DER bytes. Otherwise, return der_cert_data unchanged.
+    """
+    if not der_cert_data or der_cert_data[0] != 0x04:
+        return der_cert_data
+
+    try:
+        octet_string, _ = decoder.decode(der_cert_data, asn1Spec=univ.OctetString())
+        unwrapped = bytes(octet_string)
+    except Exception as err:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)